### PR TITLE
[posix] add a vendor interface for the radio spinel layer

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit adds a vendor interface for developers to implement
the bus interface by themselves.

### CODE REVIEW

**Review by abtink — APPROVED** [review]
👍

**Review by bukepo — COMMENTED** [review]
could you add a dummy interface for testing?

**Comment by zhanglongxia** [comment]
> could you add a dummy interface for testing?



> could you add a dummy interface for testing?

Done

**Review by bukepo — APPROVED** [review]

File: src/posix/platform/vendor_interface.hpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+     * @retval OT_ERROR_NO_BUFS  Insufficient buffer space available to encode the frame.
+     * @retval OT_ERROR_FAILED   Failed to call the SPI driver to send the frame.
+     *
+     */
+    otError SendFrame(const uint8_t *aFrame, uint16_t aLength);
+
+    /**
+     * This method waits for receiving part or all of spinel frame within specified interval.
+     *
+     * @param[in] aTimeout  The timeout value in microseconds.
```
Comment (jwhui) [inline src/posix/platform/vendor_interface.hpp:111]:
```suggestion
     * @param[in] aTimeoutUs  The timeout value in microseconds.
```

Comment (zhanglongxia) [reply src/posix/platform/vendor_interface.hpp]: Updated

**Review by jwhui — APPROVED** [review]

**Pull request merged**